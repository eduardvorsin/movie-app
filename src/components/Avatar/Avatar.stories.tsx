import type { Meta, StoryObj } from '@storybook/react';
import Avatar, { Props } from './Avatar';
import AvatarPicture from '../../assets/avatar-example.jpg';

type Statuses = Exclude<Props['status'], undefined>;
type Sizes = Exclude<Props['size'], undefined>;
type Presences = Exclude<Props['presence'], undefined>;

type Variant = {
	id: number,
	appearance?: Props['appearance'],
	isDisabled?: Props['isDisabled'],
	initials?: Props['initials'],
	presence?: Presences,
	status?: Statuses,
	src?: Props['src'],
};

const statuses: Statuses[] = ['approved', 'declined', 'locked'];
const sizes: Sizes[] = ['xsmall', 'small', 'medium', 'large', 'xlarge', 'xxlarge'];
const presences: Presences[] = ['offline', 'online'];
const variants: Variant[] = [
	{
		id: 0,
		appearance: 'circle'
	},
	{
		id: 1,
		appearance: 'square'
	},
	{
		id: 2,
		src: AvatarPicture.src
	},
	{
		id: 3,
		initials: 'AS'
	},
	{
		id: 4,
		isDisabled: true
	},
	{
		id: 5,
		status: 'approved'
	},
	{
		id: 6,
		status: 'declined'
	},
	{
		id: 7,
		status: 'locked'
	},
	{
		id: 8,
		presence: 'offline'
	},
	{
		id: 9,
		presence: 'online'
	},
];

const meta: Meta<typeof Avatar> = {
	title: 'components/Avatar',
	tags: ['autodocs'],
	argTypes: {
		appearance: {
			description: 'Appearance for Avatar',
		},
		className: {
			description: 'Additional classes for Avatar'
		},
		label: {
			description: 'Text description for screen readers',
		},
		onClick: {
			control: false,
			description: 'The callback is performed by clicking on the avatar'
		},
		size: {
			description: 'Disabling an avatar makes it non-functional',
		},
		initials: {
			description: 'Initials that are displayed if there is no picture or icon of the user',
		},
		src: {
			description: 'Image url',
		},
		presence: {
			description: 'Indicates whether the user is online or offline',
		},
		status: {
			description: 'Displays the user\'s status',
		},
		isDisabled: {
			description: 'Disabling an avatar makes it non-functional',
		},
		testId: {
			description: 'Id for testing the component',
		}
	},
	component: Avatar,
	parameters: {
		docs: {
			description: {
				component: 'An Avatar serves as a visual depiction of a user, group, or entity. It has the capability to exhibit an image, icon, or initials, and is adaptable to different dimensions and forms.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Avatar>;

export const Default: Story = {
	args: {
		label: 'default',
	},
	parameters: {
		docs: {
			description: {
				story: 'Default state if initials or url for the image are not specified',
			},
		},
	},
};

export const Circle: Story = {
	args: {
		appearance: 'circle',
	},
	parameters: {
		docs: {
			description: {
				story: 'Allows you to set a round shape for the avatar',
			},
		},
	},
};

export const Square: Story = {
	args: {
		appearance: 'square',
	},
	parameters: {
		docs: {
			description: {
				story: 'Allows you to set a square shape for the avatar',
			},
		},
	},
};

export const Disabled: Story = {
	args: {
		isDisabled: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Apply this to actions that are currently unavailable. The user interface should effectively convey why the avatar is inactive and what steps are needed to activate it.',
			},
		},
	},
};

export const Initials: Story = {
	args: {
		initials: 'AS',
	},
	parameters: {
		docs: {
			description: {
				story: 'Allows you to show initials inside the avatar if the picture is not loaded',
			},
		},
	},
};

export const Image: Story = {
	args: {
		initials: 'AS',
		src: AvatarPicture.src,
		size: 'xxlarge',
	},
	parameters: {
		docs: {
			description: {
				story: 'Avatar with a picture',
			},
		},
	},
};

export const Statuses: Story = {
	parameters: {
		docs: {
			description: {
				story: 'Allows you to set the user status. By default, the status is not displayed. There are 3 statuses in total: approved, declined, locked.',
			},
		},
	},
	render: () => (
		<div
			className='flex items-start'
		>
			{statuses.map((status, index) => (
				<Avatar
					className='mr-3 last:mr-3'
					key={status}
					label={`avatar status variant ${index + 1}`}
					size='xlarge'
					status={status}
				/>
			))}
		</div>
	),
};

export const Pressence: Story = {
	parameters: {
		docs: {
			description: {
				story: 'Allows you to set the user\'s presence status. By default, the presence status is not displayed. There are 2 values available - offline and online.',
			},
		},
	},
	render: () => (
		<div
			className='flex items-start'
		>
			{presences.map((presence, index) => (
				<Avatar
					className='mr-3 last:mr-3'
					key={presence}
					label={`avatar presence variant ${index + 1}`}
					size='xlarge'
					presence={presence}
				/>
			))}
		</div>
	),
};

export const EventCallbacks: Story = {
	argTypes: {
		onClick: {
			action: 'Clicked',
		},
	},
	args: {
		label: 'click event'
	},
	parameters: {
		docs: {
			description: {
				story: 'Callback function can be supplied to onClick event',
			},
		},
	},
};

export const Sizes: Story = {
	parameters: {
		docs: {
			description: {
				story: 'Allows you to set the size of the avatar. Medium is set by default. There are 7 sizes in total, 6 sizes with an explicit width and height, and the 7th custom option in which you need to set the sizes yourself through classes.',
			},
		},
	},
	render: () => (
		<div
			className='flex items-start'
		>
			{sizes.map((size, index) => (
				<Avatar
					className='mr-3 last:mr-3'
					key={size}
					label={`avatar size variant ${index + 1}`}
					size={size}
				/>
			))}
		</div>
	),
};

export const All: Story = {
	parameters: {
		docs: {
			description: {
				story: 'Here are all possible variants of avatars, in size, statuses, states, shape, etc.',
			},
		},
	},
	render: () => (
		<div
			className='grid gap-y-6 items-start'
		>
			{sizes.map((size) => (
				<div className='flex flex-wrap items-start'>
					{variants.map((variant) => (
						<Avatar
							className='mr-4 last:mr-0 mb-4 last:mb-0'
							appearance={variant.appearance}
							key={`${variant.id}-${size}`}
							isDisabled={variant.isDisabled}
							initials={variant.initials}
							src={variant.src}
							presence={variant.presence}
							status={variant.status}
							size={size}
							label={`avatar size ${variant} ${variant.id + 1}`}
						/>
					))}
				</div>
			))}
		</div>
	),
};