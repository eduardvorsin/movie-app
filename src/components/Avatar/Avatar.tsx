'use client';
import Image, { ImageProps } from 'next/image';
import { MouseEventHandler } from 'react';
import Approved from '../../assets/icons/approved.svg?url';
import Declined from '../../assets/icons/declined.svg?url';
import Locked from '../../assets/icons/locked.svg?url';
import { GeneralProps } from '@/types/shared';

export type Props = {
	appearance?: 'circle' | 'square',
	isDisabled?: boolean,
	label: string,
	onClick?: MouseEventHandler<HTMLElement>,
	size?: 'small' | 'xsmall' | 'medium' | 'large' | 'xlarge' | 'xxlarge',
	initials?: string,
	src?: string,
	presence?: 'offline' | 'online',
	status?: 'approved' | 'declined' | 'locked',
	imgProps?: Omit<ImageProps, 'src' | 'alt'>,
} & GeneralProps;

const statusIcons = {
	approved: (
		<svg className='fill-green-400 dark:fill-green-300' viewBox='0 0 20 20'>
			<use href={`${Approved.src}#approved`}></use>
		</svg>
	),
	declined: (
		<svg className='fill-red-400 dark:fill-red-500' viewBox='0 0 20 20'>
			<use href={`${Declined.src}#declined`}></use>
		</svg>
	),
	locked: (
		<svg className='fill-neutral-0 dark:fill-dark-neutral-0' viewBox='0 0 20 20'>
			<use href={`${Locked.src}#locked`}></use>
		</svg>
	),
} as const;

const presencePositions = {
	square: {
		xsmall: 'after:-bottom-[10%] after:-right-[10%]',
		small: 'after:-bottom-[11%] after:-right-[11%]',
		medium: 'after:-bottom-[9%] after:-right-[9%]',
		large: 'after:-bottom-[10%] after:-right-[10%]',
		xlarge: 'after:-bottom-[4%] after:-right-[4%]',
		xxlarge: 'after:-bottom-[2%] after:-right-[2%]',
	},
	circle: {
		xsmall: 'after:-bottom-[12%] after:-right-[12%]',
		small: 'after:-bottom-[10%] after:-right-[10%]',
		medium: 'after:-bottom-[8%] after:-right-[8%]',
		large: 'after:-bottom-[4%] after:-right-[4%]',
		xlarge: 'after:bottom-[2%] after:right-[2%]',
		xxlarge: 'after:bottom-[4%] after:right-[4%]',
	}
} as const;

const statusPositions = {
	square: {
		xsmall: '',
		small: '',
		medium: '-top-[4%] -right-[4%]',
		large: '-top-[6%] -right-[6%]',
		xlarge: '-top-[2%] -right-[2%]',
		xxlarge: '-top-[1%] -right-[1%]',
	},
	circle: {
		xsmall: '',
		small: '',
		medium: '-top-[6%] -right-[6%]',
		large: '-top-[4%] -right-[4%]',
		xlarge: 'top-[4%] right-[4%]',
		xxlarge: 'top-[6%] right-[6%]',
	}
} as const;

const sizeTypes = {
	xsmall: 'w-4 h-4 text-[0.5rem] border-[0.0625rem] after:w-[0.375rem] after:h-[0.375rem] after:border-[0.0625rem]',
	small: 'w-6 h-6 text-75 border-[0.09375rem] after:w-[0.625rem] after:h-[0.625rem] after:border-[0.09375rem]',
	medium: 'w-8 h-8 text-200 border-[0.09375rem] after:w-[0.75rem] after:h-[0.75rem] after:border-2',
	large: 'w-10 h-10 text-300 border-2 after:w-[0.9375rem] after:h-[0.9375rem] after:border-2',
	xlarge: 'w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2',
	xxlarge: 'w-32 h-32 text-[4rem] border-2 after:w-[1.25rem] after:h-[1.25rem] after:border-2',
} as const;

const statusSizesTypes = {
	xsmall: 'w-0 h-0',
	small: 'w-0 h-0',
	medium: 'w-[0.75rem] h-[0.75rem]',
	large: 'w-[0.9375rem] h-[0.9375rem]',
	xlarge: 'w-[1.125rem] h-[1.125rem]',
	xxlarge: 'w-[1.25rem] h-[1.25rem]',
} as const;

const borderRadiusTypes = {
	xsmall: 'rounded-0.5',
	small: 'rounded-0.5',
	medium: 'rounded-[0.1875rem]',
	large: 'rounded-[0.1875rem]',
	xlarge: 'rounded-[0.375rem]',
	xxlarge: 'rounded-3',
} as const;

export default function Avatar({
	className,
	appearance = 'circle',
	isDisabled,
	label,
	onClick,
	presence,
	size = 'medium',
	initials,
	src,
	status,
	testId,
	imgProps,
	...props
}: Props) {
	const classes = [
		'flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250',
		appearance === 'circle' ? 'rounded-full' : borderRadiusTypes[size],
		sizeTypes[size],
		isDisabled ? 'opacity-disabled cursor-not-allowed' : '',
		!src ? 'bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700' : 'bg-transparent border-transparent',
		presence ? `after:absolute after:border-light-100 dark:after:border-dark-neutral-250 after:rounded-full ${presencePositions[appearance][size]}` : 'after:hidden',
		presence === 'online' ? 'after:bg-green-600 dark:after:bg-green-500' : '',
		presence === 'offline' ? 'after:bg-red-600 dark:after:bg-red-500' : '',
		className,
	].join(' ');

	const imageClasses = [
		'object-cover',
		appearance === 'circle' ? 'rounded-full' : borderRadiusTypes[size],
	].join(' ');

	const statusWrapperClasses = [
		'w-5 h-5 absolute z-100',
		appearance === 'circle' ? 'bg-neutral-700 dark:bg-dark-neutral-700 rounded-full' : '',
		statusPositions[appearance][size],
		statusSizesTypes[size],
	].join(' ');

	const IconOrInitials = !initials ?
		(<svg
			className='w-full h-full fill-current select-none'
			viewBox='0 0 40 40'
			aria-hidden
		>
			<path d='M8.28 27.5A14.95 14.95 0 0120 21.8c4.76 0 8.97 2.24 11.72 5.7a14.02 14.02 0 01-8.25 5.91 14.82 14.82 0 01-6.94 0 14.02 14.02 0 01-8.25-5.9zM13.99 12.78a6.02 6.02 0 1112.03 0 6.02 6.02 0 01-12.03 0z'></path>
		</svg>)
		:
		initials;

	const labelText = `${label} ${status ?? ''} ${presence ?? ''}`;

	return (
		// eslint-disable-next-line jsx-a11y/click-events-have-key-events
		<span
			className={classes}
			role={!src ? 'img' : undefined}
			aria-label={!src ? labelText : undefined}
			onClick={onClick}
			data-testid={testId}
			{...props}
		>
			{(size !== 'xsmall' && size !== 'small' && status) && (
				<span
					className={statusWrapperClasses}
				>
					{statusIcons[status]}
				</span>
			)}
			{!src ?
				IconOrInitials
				:
				(
					<Image
						className={imageClasses}
						src={src}
						alt={labelText}
						fill
						{...imgProps}
					/>
				)
			}
		</span>
	);
};