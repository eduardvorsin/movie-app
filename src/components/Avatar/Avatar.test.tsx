import { render, screen } from "@testing-library/react";
import Avatar, { Props } from "./Avatar";
import userEvent from "@testing-library/user-event";
import AvatarPicture from '../../assets/avatar-example.jpg';

const appearances: NonNullable<Props['appearance']>[] = ['circle', 'square'];
const statuses: NonNullable<Props['status']>[] = ['approved', 'declined', 'locked'];
const sizes: NonNullable<Props['size']>[] = ['xsmall', 'small', 'medium', 'large', 'xlarge', 'xxlarge'];
const presences: NonNullable<Props['presence']>[] = ['offline', 'online'];

describe('Avatar tests', () => {
	it('is rendered correctly', () => {
		render(
			<Avatar
				label='avatar label'
				size='xlarge'
				testId='test-avatar'
			/>
		);

		expect(screen.getByTestId<HTMLSpanElement>('test-avatar')).toBeInTheDocument();
	});

	it('when clicking on the avatar, the mock function should be called', async () => {
		const user = userEvent.setup();
		const mockFn = jest.fn();
		render(
			<Avatar
				label='avatar label'
				size='xlarge'
				onClick={mockFn}
				testId='test-avatar'
			/>
		);

		await user.click(screen.getByTestId<HTMLSpanElement>('test-avatar'));

		expect(mockFn).toHaveBeenCalledTimes(1);
	});

	it('is a basic snapshot', () => {
		render(
			<Avatar
				label='avatar label'
				size='xlarge'
				testId='test-avatar'
			/>
		);

		expect(screen.getByTestId<HTMLSpanElement>('test-avatar')).toMatchSnapshot();
	});

	it('is a snapshot with a picture instead of initials', () => {
		render(
			<Avatar
				label='avatar label'
				size='xlarge'
				testId='test-avatar'
				src={AvatarPicture.src}
			/>
		);

		expect(screen.getByTestId<HTMLSpanElement>('test-avatar')).toMatchSnapshot();
	});

	it.each(appearances)('is a snapshot with appearance equal to "%s"', (appearance) => {
		render(
			<Avatar
				label='avatar label'
				size='xlarge'
				testId='test-avatar'
				appearance={appearance}
			/>
		);

		expect(screen.getByTestId<HTMLSpanElement>('test-avatar')).toMatchSnapshot();
	});

	it.each(statuses)('is a snapshot with status equal to "%s"', (status) => {
		render(
			<Avatar
				label='avatar label'
				size='xlarge'
				testId='test-avatar'
				status={status}
			/>
		);

		expect(screen.getByTestId<HTMLSpanElement>('test-avatar')).toMatchSnapshot();
	});

	it.each(presences)('is a snapshot with status equal to "%s"', (presence) => {
		render(
			<Avatar
				label='avatar label'
				size='xlarge'
				testId='test-avatar'
				presence={presence}
			/>
		);

		expect(screen.getByTestId<HTMLSpanElement>('test-avatar')).toMatchSnapshot();
	});

	it.each(sizes)('is a snapshot with status equal to "%s"', (size) => {
		render(
			<Avatar
				label='avatar label'
				testId='test-avatar'
				size={size}
			/>
		);

		expect(screen.getByTestId<HTMLSpanElement>('test-avatar')).toMatchSnapshot();
	});
});