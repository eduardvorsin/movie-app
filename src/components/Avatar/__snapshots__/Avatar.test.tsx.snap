// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar tests is a basic snapshot 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with a picture instead of initials 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with appearance equal to "circle" 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with appearance equal to "square" 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-[0.375rem] w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "approved" 1`] = `
<span
  aria-label="avatar label approved "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <span
    class="w-5 h-5 absolute z-100 transition-colors duration-150 bg-neutral-700 dark:bg-dark-neutral-700 rounded-full top-[4%] right-[4%] w-[1.125rem] h-[1.125rem]"
  >
    <svg
      class="fill-green-400 dark:fill-green-300 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/approved.svg#approved"
      />
    </svg>
  </span>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "declined" 1`] = `
<span
  aria-label="avatar label declined "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <span
    class="w-5 h-5 absolute z-100 transition-colors duration-150 bg-neutral-700 dark:bg-dark-neutral-700 rounded-full top-[4%] right-[4%] w-[1.125rem] h-[1.125rem]"
  >
    <svg
      class="fill-red-400 dark:fill-red-500 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/declined.svg#declined"
      />
    </svg>
  </span>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "large" 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-10 h-10 text-300 border-2 after:w-[0.9375rem] after:h-[0.9375rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "locked" 1`] = `
<span
  aria-label="avatar label locked "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <span
    class="w-5 h-5 absolute z-100 transition-colors duration-150 bg-neutral-700 dark:bg-dark-neutral-700 rounded-full top-[4%] right-[4%] w-[1.125rem] h-[1.125rem]"
  >
    <svg
      class="fill-neutral-0 dark:fill-dark-neutral-0 transition-colors duration-150"
      viewBox="0 0 20 20"
    >
      <use
        href="/assets/icons/locked.svg#locked"
      />
    </svg>
  </span>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "medium" 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-8 h-8 text-200 border-[0.09375rem] after:w-[0.75rem] after:h-[0.75rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "offline" 1`] = `
<span
  aria-label="avatar label  offline"
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:absolute after:border-light-100 dark:after:border-dark-neutral-250 after:rounded-full after:bottom-[2%] after:right-[2%]  after:bg-red-600 dark:after:bg-red-500 "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "online" 1`] = `
<span
  aria-label="avatar label  online"
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:absolute after:border-light-100 dark:after:border-dark-neutral-250 after:rounded-full after:bottom-[2%] after:right-[2%] after:bg-green-600 dark:after:bg-green-500  "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "small" 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-6 h-6 text-75 border-[0.09375rem] after:w-[0.625rem] after:h-[0.625rem] after:border-[0.09375rem]  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "xlarge" 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-24 h-24 text-[3rem] border-2 after:w-[1.125rem] after:h-[1.125rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "xsmall" 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-4 h-4 text-[0.5rem] border-[0.0625rem] after:w-[0.375rem] after:h-[0.375rem] after:border-[0.0625rem]  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;

exports[`Avatar tests is a snapshot with status equal to "xxlarge" 1`] = `
<span
  aria-label="avatar label  "
  class="flex items-center font-regular justify-center relative text-neutral-0 dark:text-dark-neutral-0 dark:border-dark-neutral-250 transition-colors duration-150 rounded-full w-32 h-32 text-[4rem] border-2 after:w-[1.25rem] after:h-[1.25rem] after:border-2  bg-neutral-700 dark:bg-dark-neutral-700 border-neutral-700 after:hidden   "
  data-testid="test-avatar"
  role="img"
>
  <svg
    aria-hidden="true"
    class="w-full h-full fill-current select-none"
    viewBox="0 0 40 40"
  >
    <use
      href="/assets/icons/user.svg#user"
    />
  </svg>
</span>
`;
